webapp/src/tutorialCodeValidation.tsx

```diff
@@ -60,42 +60,52 @@ export class ShowValidationMessage extends data.Component<TutorialCodeValidation
             const prev = this.props.ruleComponents.find(r => r.ruleName == rule.ruleName);
             // Update if the list of blockids is different
             if ((rule.blockIds && rule.blockIds.sort().toString() != prev?.blockIds?.sort()?.toString())
-                || rule.blockIds.length != nextState.ruleSnippets?.[rule.ruleName]?.length) {
-                this.getRuleSnippetsAsync(rule);
+                || rule.blockIds.length != nextState.ruleBlocks?.[rule.ruleName]?.length) {
```

Comment:
small nit and you don't have to fix this. I can address this change when I create a PR.
```suggestion
                || rule.blockIds?.length != nextState.ruleBlocks?.[rule.ruleName]?.length) {
```